processor/src/trace/stack/tests.rs

```diff
@@ -0,0 +1,97 @@
+        Operation::U32add, // no shift, clk 1
+        Operation::Pad,    // right shift, clk 2
+        Operation::Pad,    // right shift, clk 3
+        Operation::U32add, // no shift, clk 4
+        Operation::Drop,   // left shift, clk 5
+        Operation::Pad,    // right shift, clk 6
+        Operation::Drop,   // left shift, clk 7
+        Operation::Drop,   // left shift, clk 8
+        Operation::Drop,   // left shift, clk 9
+        Operation::Pad,    // right shift, clk 10
+        Operation::Drop,   // left shift, clk 11
+    ];
+    let init_stack = (1..17).collect::<Vec<_>>();
+    let mut trace = build_trace_from_ops(ops.clone(), &init_stack);
+    let alphas = rand_array::<Felt, AUX_TRACE_RAND_ELEMENTS>();
+    let aux_columns = trace.build_aux_segment(&[], &alphas).unwrap();
+    let p1 = aux_columns.get_column(P1_COL_IDX);
+
+    let row_values = [
+        OverflowTableRow::new(Felt::new(1), 2, ZERO).to_value(&alphas),
+        OverflowTableRow::new(Felt::new(2), 3, Felt::new(2)).to_value(&alphas),
+        OverflowTableRow::new(Felt::new(2), 6, Felt::new(2)).to_value(&alphas),
+        OverflowTableRow::new(ZERO, 10, ZERO).to_value(&alphas),
+    ];
+
+    // make sure the first entry is ONE
+    let mut expected_value = ONE;
+    assert_eq!(expected_value, p1[0]);
+
+    // SPAN and U32ADD do not affect the overflow table
+    assert_eq!(expected_value, p1[1]);
+    assert_eq!(expected_value, p1[2]);
+
+    // two PADs push values 1 and 2 onto the overflow table
+    expected_value *= row_values[0];
+    assert_eq!(expected_value, p1[3]);
+    expected_value *= row_values[1];
+    assert_eq!(expected_value, p1[4]);
+
+    // U32ADD does not affect the overflow table
+    assert_eq!(expected_value, p1[5]);
+
+    // DROP removes a row from the overflow table
+    expected_value *= row_values[1].inv();
+    assert_eq!(expected_value, p1[6]);
+
+    // PAD pushes the value onto the overflow table again
+    expected_value *= row_values[2];
+    assert_eq!(expected_value, p1[7]);
+
+    // two DROPs remove both values from the overflow table
+    expected_value *= row_values[2].inv();
+    assert_eq!(expected_value, p1[8]);
+    expected_value *= row_values[0].inv();
+    assert_eq!(expected_value, p1[9]);
+
+    // at this point the table should be empty
+    assert_eq!(expected_value, ONE);
+
+    // the 3rd DROP should not affect the overflow table as it is already empty
+    assert_eq!(expected_value, p1[10]);
+
+    // PAD pushes the value (ZERO) onto the overflow table again
+    expected_value *= row_values[3];
+    assert_eq!(expected_value, p1[11]);
+
+    // and then the last DROP removes it from the overflow table
+    expected_value *= row_values[3].inv();
+    assert_eq!(expected_value, p1[12]);
+
+    // at this point the table should be empty again, and it should stay empty until the end
+    assert_eq!(expected_value, ONE);
+    for i in 13..(p1.len() - NUM_RAND_ROWS) {
+        assert_eq!(ONE, p1[i]);
+    }
```

Comment:
Fixed! How do you run it to make sure it checks the test code too? I didn't realize it wasn't doing this.

Reply:
We can add the "--tests" flag to make it check tests as well. Also, apparently "--all" is now deprecated in favor of "--workspace", and I think we don't need both `-D clippy::all` and `-D warnings` since `warnings` will include warnings from clippy. `-D clippy::all` is useful if we want to fail only on warnings from clippy but not on other rustc warnings, such as `dead_code`. (This is based on the note about `-D warnings` in the [clippy repo](https://github.com/rust-lang/rust-clippy) and a quick test I ran).

Therefore we should be able to update to the following:
`cargo clippy --workspace --tests -- -D warnings`

Reply:
Thank you! So, I guess the full check the we run could be:
```
cargo fix --allow-staged --allow-dirty --all-targets --all-features; cargo fmt; cargo clippy --workspace --tests -- -D warnings
```
right?